chat/src/main.rs

```diff
@@ -0,0 +1,132 @@
+use image::ImageFormat;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{
+    env,
+    fs::{canonicalize, read},
+    io::{self, Write},
+    path::PathBuf,
+    thread,
+};
+use wry::application::window::{Icon, Window};
+use wry::{
+    application::{
+        event::{Event, StartCause, WindowEvent},
+        event_loop::{ControlFlow, EventLoop},
+        window::WindowBuilder,
+    },
+    webview::WebViewBuilder,
+};
+
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "command", content = "data")]
+enum Message {
+    #[serde(rename = "focus")]
+    Focus,
+    #[serde(rename = "set_always_on_top")]
+    SetOnTop(bool),
+}
+
+const WINDOW_TITLE: &str = "Tabnine Chat";
+
+const BASE_URL: &str = "wry://localhost";
+
+static INDEX_HTML: Lazy<String> = Lazy::new(|| {
+    let index_html = read(PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("./index.html")).unwrap();
+    Regex::new("(href|src)=\"/static")
+        .unwrap()
+        .replace_all(
+            &String::from_utf8(index_html).unwrap(),
+            format!("$1=\"{BASE_URL}/static"),
+        )
+        .to_string()
```

Comment:
I'm moving all the `tabnine-vscode` assets to `/chat/assets/*`. Do I need to change anything here except for `./index.html` to `./assets/index.html`?

Reply:
Nevermind, I realized I needed to change `/static` to `/assets/static`. I do have to wonder why you don't change {BASE_URL} to refer to the `/assets` folder though (if possible?)